tests/unit/test_example_data.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+
+import pytest
+
+from ehrql.file_formats.csv import CSVRowsReader
+from ehrql.query_model.column_specs import get_column_specs_from_schema
+from ehrql.query_model.introspection import get_table_nodes
+from ehrql.tables import core, tpp
+
+
+filenames = [
+    "addresses.csv",
+    "clinical_events.csv",
+    "medications.csv",
+    "ons_deaths.csv",
+    "patients.csv",
+    "practice_registrations.csv",
+]
+
+ql_tables = [
+    tpp.addresses,
+    tpp.clinical_events,
+    tpp.medications,
+    core.ons_deaths,
+    core.patients,
+    tpp.practice_registrations,
```

Comment:
It's not obvious where this particular list of tables comes from. That's not a reflection on your code! I just don't think we've been systematic in deciding what tables we're providing example data for. A reasonable approach would be: everything in `core` and every table used in the tutorial.

But whatever we decide, we should be dynamically constructing the list of tables here otherwise someone could add a new core table, or add a table to the tutorial, and nothing would tell them that they had failed to add it to the example data.